src/runtime/format.c

```diff
@@ -172,7 +172,7 @@ KLIB_EXPORT(rsnprintf);
 void rprintf(const char *format, ...)
 {
     /* What's a reasonable limit here? This needs to be reentrant. */
-    buffer b = little_stack_buffer(1024);
+    buffer b = little_stack_buffer(32768);
```

Comment:
This causes a stack overflow when executing in an exception handler on x86, where exception handlers have a 32 KB stack, which means that when an unhandled fault occurs, the register dump and frame and stack traces that are supposed to be printed on the console are often replaced by those associated to this stack overflow, with a generic "frame xxxx already full" message that hides the original fault.
Would it be OK to decrease the rprintf buffer size here to, say, 16 KB?

Reply:
> Would it be OK to decrease the rprintf buffer size here to, say, 16 KB?

Sure. This change shouldn't have been in this merge - but I guess I let it slip as it seemed rather frequent that a "!buffer_is_wrapped" assertion would be hit while debugging and printing >1KB of output (e.g. when printing the full content of a tuple or buffer). 32KB was an arbitrary choice.

While we're at it, we may as well increase the size of the exception handler stack - I can't think of any reason why it needs to be constrained to 32KB. All these numbers are arbitrary - we just need to make sure things don't fall over if someone breathes on it the wrong way.

Reply:
#1478